cds/modules/deposit/static/js/cds_deposit/avc/components/cdsDeposit.js

```diff
@@ -166,14 +166,28 @@ function cdsDepositCtrl(
     };
 
     this.videoPreviewer = function(deposit, key) {
-      if (that.record._files[0] &&
-        (that.stateQueue.SUCCESS.indexOf('file_download') > -1 || key)) {
-        that.previewer = $sce.trustAsResourceUrl(
-          urlBuilder.video({
-            deposit: deposit || that.record._deposit.id,
-            key: key || that.record._files[0].key,
-          })
-        );
+      var videoUrl;
+      if (deposit && key) {
+        videoUrl = urlBuilder.video({
+          deposit: deposit,
+          key: key,
+        });
+      } else {
+        var videoFile = that.record._files[0];
```

Comment:
This is making the assumption that the first file in the list is going to be always the video, and it doesn't have to be always like that.
We have added `content_type = 'video'` as a subfield for the master file too, maybe you can filter all files in the first level and get only the one with this content type; we know for sure there is only one master video per record.